### PR TITLE
Admin: Search All Datasets button

### PR DESCRIPTION
**Pull request by david-nishi**
## Major Changes
- Admin users have access to a search button that allows for private datasets.
- When searching all datasets, filtering by dataset types (public, follow, my datasets, bookmarked) and "My Dataset Status" options are not available.

### CODE REVIEW

File: packages/openneuro-app/src/scripts/refactor_2021/search/inputs/admin-allDatasets-toggle.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import React, { FC, useContext } from 'react'
+import { SearchParamsCtx } from '../search-params-ctx'
+import { Button } from '@openneuro/components/button'
+import styled from '@emotion/styled'
+
+const StyledButton = styled(Button)`
+  width: 100%;
+  padding: 6px 10px;
+  margin-bottom: 20px;
+`
```
Comment (thinknoack) [inline packages/openneuro-app/src/scripts/refactor_2021/search/inputs/admin-allDatasets-toggle.tsx:19]:
```suggestion
&.active{
    background: #fff5f3;
    border-color: #e68383;
    color: #710000;
    }
`
```

Comment (thinknoack) [reply packages/openneuro-app/src/scripts/refactor_2021/search/inputs/admin-allDatasets-toggle.tsx:19]: Button Component will need an active class. Unsure if emotion uses the & to concat classNames.

File: packages/openneuro-app/src/scripts/refactor_2021/search/inputs/admin-allDatasets-toggle.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  const toggleSearchAllDatasets = () =>
+    setSearchParams(prevState => ({
+      ...prevState,
+      searchAllDatasets: !prevState.searchAllDatasets,
+    }))
+
+  return (
+    <StyledButton
+      className="toggle-btn active"
+      onClick={toggleSearchAllDatasets}
```
Comment (thinknoack) [inline packages/openneuro-app/src/scripts/refactor_2021/search/inputs/admin-allDatasets-toggle.tsx:36]:
```suggestion
      onClick={toggleSearchAllDatasets}
      icon={searchAllDatasets ? "fas fa-check-square" : "far fa-square" }
```

Comment (thinknoack) [reply packages/openneuro-app/src/scripts/refactor_2021/search/inputs/admin-allDatasets-toggle.tsx:36]: maybe use an active/inactive icon?

File: packages/openneuro-app/src/scripts/refactor_2021/search/inputs/admin-allDatasets-toggle.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+  const toggleSearchAllDatasets = () =>
+    setSearchParams(prevState => ({
+      ...prevState,
+      searchAllDatasets: !prevState.searchAllDatasets,
+    }))
+
+  return (
+    <StyledButton
+      className="toggle-btn active"
```
Comment (thinknoack) [inline packages/openneuro-app/src/scripts/refactor_2021/search/inputs/admin-allDatasets-toggle.tsx]:
```suggestion
      className={searchAllDatasets ? "active toggle-btn" : "toggle-btn"}
```

**Review by thinknoack — COMMENTED** [review]
works well. I may want to futz with the styling later but I added some suggestions for now.

File: packages/openneuro-app/src/scripts/refactor_2021/search/inputs/admin-allDatasets-toggle.tsx (hunk 1 above)
Comment (nellh) [reply packages/openneuro-app/src/scripts/refactor_2021/search/inputs/admin-allDatasets-toggle.tsx:19]: > Button Component will need an active class. Unsure if emotion uses the & to concat classNames.

It is supported in Emotion.

**Review by thinknoack — APPROVED** [review]

**Pull request merged**